code/create/dta_dicom2spec_rules.py

```diff
@@ -215,10 +160,36 @@ def _rules(self, series_dict, subject=None, anon_subject=None,
                 'bids-task': apply_bids_label_restrictions(_guess_task(self, series_dict) ),
                 'id': series_dict.get('SeriesNumber', 'unknown'),
 		'bids-run' : apply_bids_label_restrictions( run )
-              #  'bids-run' : apply_bids_label_restrictions(_guess_run(self, series_dict)) if "ProtocolName" in series_dict else None
                 }
 
+    # The table contains rows like "DTA003   T12122   T15722"
+    # so if you find a T-value in the table, you know, that the
+    # first field of that row contains the DTA003-value, that
+    # I have to use as the subject-ID
+    def get_column( self, t_value ):
+	# TODO: check if the relative path with .. in the beginning works.
+#         table_path = "/data/BnB1/DATA/source_data/DTA/code/dtage.tsv"
+         table_path = "../dtage.tsv"
+         with open( table_path, 'r' ) as table:
+            for line in table:
+               for part in line.split("\t"):
+                  if t_value in part:
+                     return str(line.split("\t").index(part))
+         return None
+
+
+    # this function reads a table and searches the given t-value (these are the session values for dta)
+    # when it finds the session value, it retunrs the matching subject-ID from the table.
+    def read_subject_for_session_from_table( self, t_value ):
+        table_path = "/data/BnB1/DATA/source_data/DTA/code/dtage.tsv"
+        with open( table_path, 'r' ) as table:
+            for line in table:
+                if t_value in line:
+                   return line.replace("\n","\t").split("\t")[0]
```

Comment:
Now, that I read it, it seems that the function above (`get_column`) is actually replaced by `read_subject_for_session_from_table`. If so - good! That's clearly more efficient. But delete the not-needed one.

As for this function itself:

1. `replace("\n", "\t")` for a line is confusing. `\n` is a line separator, so it only can appear at the end of the line. No need to go through the string looking for that character and replace it. Just remove white spaces at the end, or the beginning and the end, by calling `line.rstrip()` or `line.strip()` respectively. Or if you want to be more specific about what to strip, pass "\n" as parameter to the strip function.

2. Similar issue with `t_value in line` as above.
Be aware that `line` is a string, not a list. So, `in` matches any place within that string. It isn't necessarily a "word" or something like that. It would be correct, if `line` in that case was a list as returned by that `split('\t')`. Then `in` would match actual members of that list.

Therefore better:
```
for line in table:
    cols = line.strip().split('\t')
    if t_value in cols:
        return cols[0]
```